syntax = "proto3";

service AggregatorService {
    rpc PeristHeartBeat (PersistableHeartBeat) returns (HeartBeatPersistedACK) {}
    rpc GetAllNodesHealth(AllNodesHealthRequest) returns (AllNodesHealthResponse) {}
}

message PersistableHeartBeat {
    string node_ip = 1;
    string node_status = 2;
    string timestamp = 3;
    string response_time = 4;
}

message HeartBeatPersistedACK {
    string ACK = 1;
}


message AllNodesHealthRequest{

} 

message AllNodesHealthResponse{
    repeated PersistableHeartBeat allNodesHealth = 1;
}